src/lib/stitching/kaws/stitching.ts

```diff
@@ -117,15 +117,17 @@ export const kawsStitchingEnvironment = (
         resolve: (parent, _args, context, info) => {
           const query = parent.query
           const hasKeyword = Boolean(parent.query.keyword)
-          const contextWithCache = {
-            ...context,
-            filterArtworksLoader: loaderParams =>
-              context.filterArtworksLoader(
-                { ...loaderParams },
-                { requestThrottleMs: 1000 * 60 * 60 } // 1 hour
-              ),
+
+          const existingLoader =
+            context.unauthenticatedLoaders.filterArtworksLoader
+          const newLoader = loaderParams => {
+            return existingLoader.call(null, loaderParams, {
```

Comment:
Not that this doesn’t work, but why the `call` instead of just invoking it directly?

Reply:
When written as `existingLoader(loaderParams, ...)`, you get a max call stack error.

If you look at the code, it's kind of self-referential. The new loader calls the old loader, and then the new loader is copied into...the old loader. With a `.call` or `.apply` everything worked.

So it must be something about how invoking it that way makes a copy of the function and avoids the issue..?

Reply:
Hmm, that’s very weird and kinda makes me paranoid and wanting a better understanding. As it stands, the closure only has the old loader in scope, so the re-assignment after the closure shouldn’t matter. Could the code have been different at the time you got the call-stack error?

Reply:
@alloy feel free to pull latest and give it a shot!

Reply:
Sure. Would any usage (eg existing test coverage) lead to the error or did it require specific repro steps?

Reply:
There _was_ existing test coverage, but it had to be pended b/c it actually made HTTP requests.

So in terms of maintenance around this, probably those tests should be un-pended (and correct stubbing figured out), _or_ a more thorough comment left in the code here?

From what I recall, the `max call stack size error` occurred with this exact code, and went away when switching to `call` or `apply`. Then the tests were correctly error-ing and were fixed, when I noticed the HTTP requests and pended them. So I would think that the specs would have error-ed with `max call stack size error`, but I only ran the tests after I had it working in GraphiQL, so I didn't run the tests during that first attempt.

Reply:
I'm happy to look into stubbing these or pairing on that project as a follow up.